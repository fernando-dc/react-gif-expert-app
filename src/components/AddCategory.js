import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState("");
    
    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    };
    const handleSubmit = (e) =>{
        e.preventDefault();
        //setCategories will use the callback function to access the array elements since this component cant do that
        if (inputValue.trim().length > 2){
            setCategories( categories => [inputValue, ...categories]);
            setinputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input 
                className='animate__animated animate__fadeInDown animate__delay-1s'
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                placeholder='Input new value'
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
    