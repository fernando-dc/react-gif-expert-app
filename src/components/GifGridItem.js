import React from 'react';

export const GifGridItem = ({id, title, url}) => {
    console.log(id, title, url);
    return(
        <div className='card animate__animated animate__fadeIn animate__slow'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};
