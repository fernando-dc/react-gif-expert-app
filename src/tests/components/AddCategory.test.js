import React from 'react';
import {shallow} from 'enzyme';

import  {AddCategory} from '../../components/AddCategory';

describe('Testing for AddCategory component', () => {

    const setCategories = jest.fn();
    const wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    test('Must show component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Change in input box', () => {
        const input = wrapper.find('input');
        const value = 'Testing input';

        input.simulate('change',{ target: {value:value} });
    });

    test('Must not send POST data on submit', () => {
        const form = wrapper.find('form').simulate('submit', {});


    });
    
    
});
