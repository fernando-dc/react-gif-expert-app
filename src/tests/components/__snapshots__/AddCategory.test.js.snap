// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for AddCategory component Must show component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="animate__animated animate__fadeInDown animate__delay-1s"
    onChange={[Function]}
    placeholder="Input new value"
    type="text"
    value=""
  />
</form>
`;
